src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Elements.hpp

```diff
@@ -461,6 +481,72 @@ namespace AirflowNetwork {
         virtual ComponentType type() = 0;
     };
 
+    int constexpr NrInt = 20; // Number of intervals for a large opening
+
+    struct DetailedOpeningSolver
+    {
+        // Large opening variables
+        EPVector<Real64> DpProf;   // Differential pressure profile for Large Openings [Pa]
+        EPVector<Real64> RhoProfF; // Density profile in FROM zone [kg/m3]
+        EPVector<Real64> RhoProfT; // Density profile in TO zone [kg/m3]
+        Array2D<Real64> DpL;       // Array of stack pressures in link
+
+        void allocate(int number_of_links, int n_dop)
+        {
+            DpProf.allocate(n_dop * (NrInt + 2));
+            RhoProfF.allocate(n_dop * (NrInt + 2));
+            RhoProfT.allocate(n_dop * (NrInt + 2));
+            DpL.allocate(number_of_links, 2);
+        }
+
+        void clear()
+        {
+            DpProf.clear();
+            RhoProfF.clear();
+            RhoProfT.clear();
+            DpL.clear();
+        }
+
+        void presprofile(EnergyPlusData &state,
+                         int const il,                  // Linkage number
+                         int const Pprof,               // Opening number
+                         Real64 const G,                // gravitation field strength [N/kg]
+                         const Array1D<Real64> &DpF,    // Stack pressures at start heights of Layers
```

Comment:
`EPVector`s? `EPVector` of `struct` rather than six `EPVector`?

Reply:
Yes, I believe so, working on verifying.